conmon/conmon.c

```diff
@@ -319,8 +321,30 @@ int write_k8s_log(int fd, stdpipe_t pipe, const char *buf, ssize_t buflen)
 			}
 		}
 
+		write_len = line_len;
+
+		/*
+		 * We need to remove \r from a \r\n sequence while writing to the file.
+		 * This can be taken out once kubernetes enables OPOST in the client.
+		 */
+		if (*line_end == '\n' && line_len > 1) {
+			line_end--;
+			if (*line_end == '\r') {
+				buf[line_len - 2] = '\n';
```

Comment:
Pushed update to send \n separately and not modify the buf that we use to send to conn_sock.
PTAL.
Also, we do need to send \r\n to the client, otherwise the output keeps shifting to the right on every new line without the \r. We won't have to send \r once kube changes their terminal settings to include OPOST which takes care of this automatically. https://github.com/torvalds/linux/blob/v4.10/drivers/tty/n_tty.c#L433

Unfortunately, I don't see kube making that change for 1.6.x stable branch because that still needs to work with docker and older runc that have ONLCR enabled. In this PR, I am changing the latest runc's settings to toggle ONLCR to be compatible with kube 1.6.x